File: dura-vermeer-dashboard/Models/DB/DataPoint.cs
using System;
using System.Collections.Generic;

namespace dura_vermeer_dashboard.Models.DB
{
    public partial class DataPoint
    {
        public int DataPointId { get; set; }
        public string Naam { get; set; }
        public DateTime? Datum { get; set; }
        public int Rssi { get; set; }
        public int Afstand { get; set; }
        public int NodeId { get; set; }

        public Node Node { get; set; }
    }
}

File: dura-vermeer-dashboard/angular/app/DataPoint.ts

export class DataPoint {
  public Naam = '';
  public Datum: Date;
  public RSSI: number;
  public Afstand: number;
  public NodeId: number;


  constructor(NodeId: number) {
    this.NodeId = NodeId;
  }

  public setValue(index: number, value: string) {
    switch (index) {
      case 0:
        this.Naam = value;
        break;
      case 1:
        this.Datum = new Date(Number.parseInt(value));
        console.log('new data: ' + this.Datum);
        break;
      case 2:
        this.RSSI = Number.parseFloat(value);
        this.BerekenAfstand();
        break;
    }
  }

  private BerekenAfstand() {
    const txPower = -72.0833;
    //this.Afstand = Math.round( Math.pow(10, ((txPower - this.RSSI) / (10 * 2))) * 1000) / 1000; // in meters
    this.Afstand = Math.round( Math.pow(10, ((txPower - this.RSSI) / (10 * 2))) * 100); // in centimeters
  }

}
